util/git/creds.go

```diff
@@ -708,13 +706,10 @@ func (creds AzureWorkloadIdentityCreds) Environ() (io.Closer, []string, error) {
 	if err != nil {
 		return NopCloser{}, nil, err
 	}
-	nonce := creds.store.Add("", token)
-	env := creds.store.Environ(nonce)
 
-	return utilio.NewCloser(func() error {
-		creds.store.Remove(nonce)
-		return nil
-	}), env, nil
+	var env []string
+	env = append(env, fmt.Sprintf("%s=Authorization: Bearer %s", bearerAuthHeaderEnv, token))
```

Comment:
I don't see a reason to remove the credsStore. Perhaps the only change necessary is to do an approach similar to HTTPSCreds? https://github.com/argoproj/argo-cd/blob/c60a727524b34699669d44eff6d851174f265c65/util/git/creds.go#L243-L246